lxd/patches.go

```diff
@@ -1266,14 +1266,16 @@ func patchStorageRenameCustomISOBlockVolumesV2(name string, d *Daemon) error {
 			return fmt.Errorf("Failed loading pool %q: %w", poolName, err)
 		}
 
+		isRemotePool := p.Driver().Info().Remote
+
 		// Ensure the renaming is done only on the selected patch cluster member for remote storage pools.
-		if p.Driver().Info().Remote && !isSelectedPatchMember {
+		if isRemotePool && !isSelectedPatchMember {
 			continue
 		}
 
 		for _, vol := range volumes {
-			// In a non-clusted environment ServerName will be empty.
-			if s.ServerName != "" && vol.Location != s.ServerName {
+			// Skip volumes on local pools that are on other servers.
+			if !isRemotePool && s.ServerClustered && vol.Location != s.ServerName {
```

Comment:
I was now looking at this for a bit.

Having only `!isRemotePool && vol.Location != s.ServerName` would evaluate to `true` every time because if LXD is not clustered `s.ServerName` is empty? So therefore we need the extra check for `s.ServerClustered` (or `s.ServerName != ""` but the other is more elegant :) ).

Maybe for the future it would help to have a `isSelectedVolPatchMember(pool, vol) bool` that can be used for decision making when patching volumes?

Reply:
> if LXD is not clustered `s.ServerName` is empty?

s.ServerName is "none" in those situations.

Reply:
> Maybe for the future it would help to have a `isSelectedVolPatchMember(pool, vol) bool` that can be used for decision making when patching volumes?

Yes standardising that logic into a helper function would be useful though.

Reply:
Just thinking what will happen if you actually have a LXD cluster member called "none". I suspect this is allowed?

Maybe worth having another story that checks this too in the other places this pattern is used.

Reply:
I dont think you are allowed to do that, but if you can then it should be blocked.

Reply:
Looks to be possible:

```
root@m1:~# lxc cluster ls
+------+-----------------------------+------------------+--------------+----------------+-------------+--------+-------------------+
| NAME |             URL             |      ROLES       | ARCHITECTURE | FAILURE DOMAIN | DESCRIPTION | STATE  |      MESSAGE      |
+------+-----------------------------+------------------+--------------+----------------+-------------+--------+-------------------+
| m1   | https://10.165.133.53:8443  | database-leader  | x86_64       | default        |             | ONLINE | Fully operational |
|      |                             | database         |              |                |             |        |                   |
+------+-----------------------------+------------------+--------------+----------------+-------------+--------+-------------------+
| none | https://10.165.133.128:8443 | database-standby | x86_64       | default        |             | ONLINE | Fully operational |
+------+-----------------------------+------------------+--------------+----------------+-------------+--------+-------------------+
```

I'll open up a PR to block it.